module/motion/IKKick/src/IKKick.cpp

```diff
@@ -132,26 +128,28 @@ namespace motion {
                     supportFoot = LimbID::RIGHT_LEG;
                 }
 
-                Transform3D torsoPose = (supportFoot == LimbID::LEFT_LEG) ? leftFoot.i() : rightFoot.i();
+                Eigen::Affine3d torsoPose =
+                    (supportFoot == LimbID::LEFT_LEG) ? leftFoot.inverse() : rightFoot.inverse();
 
                 // Put the ball position from vision into torso coordinates
-                arma::vec3 targetTorso;  // = Transform3D(convert<double, 4,
-                                         // 4>(sensors.kinematicsBodyToGround)).i().transformPoint(convert<double,
-                                         // 3>(command.target)); //TODO fix
+                Eigen::Vector3d targetTorso;  // = Transform3D(convert<double, 4,
+                                              // 4>(sensors.kinematicsBodyToGround)).i().transformPoint(convert<double,
+                                              // 3>(command.target)); //TODO fix
                 // Put the ball position into support foot coordinates
-                arma::vec3 targetSupportFoot = torsoPose.transformPoint(targetTorso);
+                Eigen::Vector3d targetSupportFoot = torsoPose * targetTorso;
 
                 // Put the goal from vision into torso coordinates
-                arma::vec3 directionTorso;  // = Transform3D(convert<double, 4,
-                                            // 4>(sensors.kinematicsBodyToGround)).i().transformVector(convert<double,
-                                            // 3>(command.direction)); //TODO fix
+                Eigen::Vector3d
+                    directionTorso;  // = Transform3D(convert<double, 4,
+                                     // 4>(sensors.kinematicsBodyToGround)).i().transformVector(convert<double,
+                                     // 3>(command.direction)); //TODO fix
                 // Put the goal into support foot coordinates
-                arma::vec3 directionSupportFoot = torsoPose.transformVector(directionTorso);
+                Eigen::Vector3d directionSupportFoot = torsoPose.rotation() * directionTorso;
 
-                arma::vec3 ballPosition = targetSupportFoot;
-                ballPosition[2]         = 0.05;  // TODO: figure out why ball height is unreliable
-                arma::vec3 goalPosition = directionSupportFoot;
-                goalPosition[2]         = 0.0;  // TODO: figure out why ball height is unreliable
+                Eigen::Vector3d ballPosition = targetSupportFoot;
+                ballPosition[2]              = 0.05;  // TODO: figure out why ball height is unreliable
+                Eigen::Vector3d goalPosition = directionSupportFoot;
+                goalPosition[2]              = 0.0;  // TODO: figure out why ball height is unreliable
```

Comment:
Use `.x()`, `.y()`, `.z()` accessors whenever feasible, it will improve readability.

```suggestion
                Eigen::Vector3d ballPosition = targetSupportFoot;
                ballPosition.z()              = 0.05;  // TODO: figure out why ball height is unreliable
                Eigen::Vector3d goalPosition = directionSupportFoot;
                goalPosition.z()              = 0.0;  // TODO: figure out why ball height is unreliable
```

Reply:
Done for all motion files :+1: